src/uwtools/tests/test_cli.py

```diff
@@ -329,6 +388,29 @@ def test__dispatch_template(params):
     func.assert_called_once_with(args)
 
 
+def test__dispatch_template_render_no_optional():
+    args = {
+        STR.infile: None,
+        STR.outfile: None,
+        STR.valsfile: None,
+        STR.valsfmt: None,
+        STR.keyvalpairs: "",
```

Comment:
Was keyvalpairs supposed to be an empty list when not provided?

Reply:
The parsing that creates STR.keyvalpairs would create an empty list if not given anything, so when going back to this args{} method requires defining it as empty since we don't allow it to accept None from talking earlier

Reply:
`[]` is also empty and would be more natural to see here since that's what `argparse` would provide. We do often provide bogus and not even implausible values in tests when they'll only be received by a mock, but in this case I think that `_dict_from_key_eq_val_strings()` will actually receive and operate on this `""` value and, though it will _work_ since `""` is both iterable and empty, I think it might surprise future readers, while `[]` should be more expected.

Reply:
Interestingly I had `[]` originally but it gives `error: Need type annotation for "args"`. The typing for `config_items` in `_dict_from_key_eq_val_strings` is `List[str]`, even though the docstring reads "Strings in the form key=value"

Reply:
`args: dict = {` should do it.

Reply:
Interesting, so then the linter is mad because its not understanding the implicit typing for `[]` ? Thanks, I'll have to file that away for future reference